sql-cli/src/sql_cli/sql_directory_parser.py

```diff
@@ -0,0 +1,131 @@
+            )
+        return False
+
+    def __gt__(self, other: SqlFile) -> bool:
+        """
+        Check if this sql file path comes before the other sql file path in alphabetic order.
+
+        :param other: The other sql file to compare to.
+
+        :returns: True if this sql file path comes before the other one in alphabetic order.
+        """
+        return self.root_directory / self.path > self.root_directory / other.path
+
+    def __hash__(self) -> int:
+        """
+        Defines the hash for the sql file.
+
+        :returns: the hash of the sql file.
+        """
+        return hash(self.root_directory) ^ hash(self.path)
+
+    def get_parameters(self) -> list[str]:
+        """
+        Get all parameters used for parameterized sql queries.
+
+        :returns: declared parameters for the sql query.
+        """
+        return sorted(find_template_variables(self.path))
+
+    def has_sub_directory(self) -> bool:
+        """
+        Check if sql file is in a sub directory.
+
+        :returns: True if it's not in the parent root directory.
+        """
+        return self.path.parent != self.root_directory
+
+    def get_sub_directories(self) -> Iterable[str]:
+        """
+        Get the directory names between root and sql file path.
+
+        :yields: a sub directory name.
+        """
+        for parent in self.path.parents:
+            if parent == self.root_directory:
+                break
+            yield parent.name
+
+    def get_variable_name(self) -> str:
+        """
+        Get the variable name used as a unique identifier as a python variable.
+
+        :returns: the file name without suffix.
+        """
+        if self.has_sub_directory():
+            return f"{'__'.join(self.get_sub_directories())}__{self.path.stem}"
+        return self.path.stem
+
+    def get_relative_target_path(self) -> str:
+        """
+        Get the path to the executable sql file.
+
+        :returns: the path where sql files without any headers are being placed.
+        """
+        target_full_directory = self.target_directory / "/".join(
+            self.get_sub_directories()
+        )
+        target_full_directory.mkdir(parents=True, exist_ok=True)
+
+        target_path = target_full_directory / self.path.name
+
+        with open(target_path, "w") as file:
+            file.write(f"{self.content}\n")
+
+        return target_path.relative_to(self.root_directory.parent).as_posix()
+
+
+def get_sql_files(directory: Path, target_directory: Path) -> set[SqlFile]:
+    """
+    Get all sql files within a directory.
+
+    :param directory: The directory look in for sql files.
+    :param target_directory: The target directory path for the executable sql.
+
+    :returns: the sql files found in the directory.
+    """
+    return {
+        SqlFile(root_directory=directory, path=child, target_directory=target_directory)
+        for child in directory.rglob("*.sql")
+        if child.is_file()
```

Comment:
This is helpful to avoid cycles with symbolic links - it may be worth adding a test for this (it can be done ina  separate PR)

Reply:
This did not check for symlinks, but [b7aeae8](https://github.com/astronomer/astro-sdk/pull/836/commits/b7aeae856e5d8e7e76e08976319e4ccae7bdedf1) does.